### PR TITLE
chore: resolve `redundant_closure_for_method_calls` lint warnings

### PR DESCRIPTION
**Pull request by mehulmathur16**
# Rationale for this change

We have cargo clippy running in our CI in order to enforce code quality. In order to increase our standards, we should enable the clippy::pedantic lint group.

# What changes are included in this PR?

Resolved `redundant_closure_for_method_calls` lint warnings

# Are these changes tested?

Yes.

### CODE REVIEW

File: crates/proof-of-sql/src/base/commitment/naive_commitment.rs [hunk 1]
```diff
@@ -119,35 +119,37 @@ impl Commitment for NaiveCommitment {
...
-                        big_int_vec.iter().map(|b| b.into()).collect()
+                        big_int_vec.iter().map(std::convert::Into::into).collect()
                     }
                     CommittableColumn::Int128(int_128_vec) => {
-                        int_128_vec.iter().map(|b| b.into()).collect()
+                        int_128_vec.iter().map(std::convert::Into::into).collect()
                     }
                     CommittableColumn::Decimal75(_, _, u64_vec) => {
-                        u64_vec.iter().map(|b| b.into()).collect()
+                        u64_vec.iter().map(std::convert::Into::into).collect()
```
Comment (iajoiner) [inline crates/proof-of-sql/src/base/commitment/naive_commitment.rs:140]: We may as well import `core::convert::Into`. Also please bear in mind that we need to support `no_std` so unless it is absolutely necessary let's not use `std`.

File: crates/proof-of-sql/src/base/database/column.rs [hunk 2]
```diff
@@ -1014,8 +1011,11 @@ mod tests {
             "Spazju u Ħin",
         ];
         let scalars = strs.iter().map(Curve25519Scalar::from).collect::<Vec<_>>();
-        let owned_col =
-            OwnedColumn::VarChar(strs.iter().map(|s| s.to_string()).collect::<Vec<String>>());
+        let owned_col = OwnedColumn::VarChar(
+            strs.iter()
+                .map(std::string::ToString::to_string)
```
Comment (iajoiner) [inline crates/proof-of-sql/src/base/database/column.rs:1016]: Similarly `to_string` probably should be imported and `std` should be avoided

**Review by iajoiner — CHANGES_REQUESTED** [review]
Really thanks for your hard work, @mehulmathur16 !
Please address the comments.

**Comment by mehulmathur16** [comment]
Thanks @iajoiner.

**Review by JayWhite2357 — APPROVED** [review]

**Review by iajoiner — APPROVED** [review]
NIT: We really should put `core::convert::Into` etc at the top. I'm fine with this for now though.

**Comment by JayWhite2357** [comment]
> NIT: We really should put `core::convert::Into` etc at the top. I'm fine with this for now though.

Agreed. I'm not too opinionated on this though.

**Pull request merged**